### ISSUE TITLE
Subscribe past students

### ISSUE THREAD

**Issue by brashrebel** [realbigplugins/learndash-mailchimp#11]
Expecting this feature to be asked for. It would be nice to have a feature for one click adding of all past students to the MailChimp list. This could only be run after #8 was run. It would have to scale well though. Maybe use batch processing or some step system or a background process so it can run without failing, exhausting memory or crashing the site.

**Comment by d4mation** [comment]
Isn't this against MailChimp ToS though? I'm pretty sure they have to opt-in to any mailing lists.

**Comment by brashrebel** [comment]
Well, sure. Using the feature would technically be subscribing people we assume have not given consent. We could instead of subscribing them, trigger a double opt in email to them maybe. I'd like to find a way to capture the spirit of the issue which is to allow site owners to capture subscribers who didn't have the opportunity to subscribe on their own since the option didn't exist when they signed up.

**Comment by d4mation** [comment]
Supposedly this plugin supports this and may be worth looking into as an example.

https://mc4wp.com/premium-features/

**Comment by d4mation** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [LearnDash MailChimp - Auto-Subscribe](https://trello.com/c/Zta8UQFL/62-learndash-mailchimp-auto-subscribe)

**Comment by d4mation** [comment]
Auto-Subscribe has been implemented
